public-docs/how-to-extend-product.md

```diff
@@ -0,0 +1,112 @@
+---
+title: How To: Extend the Product Schema
+---
+
+## Prerequisite Reading
+- [Catalogs](./developer-catalog.md)
+- [Understanding Plugins](./core-plugins-intro.md)
+
+## Overview
+
+As a developer customizing Reaction, you may find a need to add some custom property to products. You should avoid this if you can achieve your goals some other way, such as using `metafields`, tags, or a separate data store that references product IDs. But in some cases, extending products is the best way.
+
+Because products have a publishing flow and have variants, extending them requires many steps. In general, they are as follows:
+- Extend database schemas
+- Extend GraphQL schemas
+- Register your custom property as published, if it should be published to the catalog
+- Register a function that publishes your custom property, if it should be published to the catalog
+- Create a GraphQL mutation for setting your custom property value
```

Comment:
The decision, at least for core, was to build new features using GraphQL and a new full-page UI experience. We can't document that quite yet because we don't have the full page route/layout created yet.

But even for existing UI, several of the core admin components, including some Blaze ones, are now using GraphQL mutations. So there are patterns in place for doing it.